ts/features/fci/components/DocumentViewer.tsx

```diff
@@ -0,0 +1,221 @@
+import React, { useState } from "react";
+import { SafeAreaView, StyleSheet } from "react-native";
+import { useNavigation } from "@react-navigation/native";
+import { constNull, pipe } from "fp-ts/lib/function";
+import * as pot from "@pagopa/ts-commons/lib/pot";
+import ReactNativeBlobUtil from "react-native-blob-util";
+import Pdf from "react-native-pdf";
+import * as S from "fp-ts/lib/string";
+import image from "../../../../img/servicesStatus/error-detail-icon.png";
+import { IOColors } from "../../../components/core/variables/IOColors";
+import WorkunitGenericFailure from "../../../components/error/WorkunitGenericFailure";
+import BaseScreenComponent from "../../../components/screens/BaseScreenComponent";
+import FooterWithButtons from "../../../components/ui/FooterWithButtons";
+import I18n from "../../../i18n";
+import IconFont from "../../../components/ui/IconFont";
+import { emptyContextualHelp } from "../../../utils/emptyContextualHelp";
+import { isIos } from "../../../utils/platform";
+import { share } from "../../../utils/share";
+import { showToast } from "../../../utils/showToast";
+import { confirmButtonProps } from "../../bonus/bonusVacanze/components/buttons/ButtonConfigurations";
+import TouchableDefaultOpacity from "../../../components/TouchableDefaultOpacity";
+import { useIODispatch, useIOSelector } from "../../../store/hooks";
+import { fciDownloadPreview, fciDownloadPreviewCancel } from "../store/actions";
+import {
+  fciDownloadPathSelector,
+  fciDownloadPreviewSelector
+} from "../store/reducers/fciDownloadPreview";
+import { LoadingErrorComponent } from "../../bonus/bonusVacanze/components/loadingErrorScreen/LoadingErrorComponent";
+import { InfoScreenComponent } from "../../../components/infoScreen/InfoScreenComponent";
+import { renderInfoRasterImage } from "../../../components/infoScreen/imageRendering";
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1
+  },
+  pdf: {
+    flex: 1,
+    backgroundColor: IOColors.bluegrey
+  }
+});
+
+const getFileNameFromUrl = (url: string) =>
+  url.substring(url.lastIndexOf("/") + 1);
+
+const renderFooter = (
+  url: string,
+  filePath: string,
+  onShare?: () => void,
+  onOpen?: () => void,
+  onDownload?: () => void
+) =>
+  isIos ? (
+    <FooterWithButtons
+      type={"SingleButton"}
+      leftButton={confirmButtonProps(() => {
+        onShare?.();
+        ReactNativeBlobUtil.ios.presentOptionsMenu(filePath);
+      }, I18n.t("features.mvl.details.attachments.pdfPreview.singleBtn"))}
+    />
+  ) : (
+    <FooterWithButtons
+      type={"ThreeButtonsInLine"}
+      leftButton={{
+        bordered: true,
+        primary: false,
+        onPress: () => {
+          onShare?.();
```

Comment:
I can't get the point of displaying a button that do nothing on press. Do we probably need to hide it?

Reply:
Ok. I'll try to remove this code. This code is a copy of the component used in PN where only the mode to get the url is changed.